File: solutions/c/secret-handshake/1/secret_handshake.c
#include "secret_handshake.h"

#include<string.h>
#include<malloc.h>

const char *actions[] = {"wink", "double blink", "close your eyes", "jump"};

const char **commands(size_t number){
   const char **result = calloc(4,sizeof(char*));
    int start = 0; int end = 4; int incr = 1;
    if (0!= (number & 16)) {start = 3; end = -1; incr = -1;}
    int idx = 0;
    for(int i=start; i != end; i += incr) {
        if(0!= (number & (1<<i))){
            result[idx++] = actions[i]; 
        }
    }
    return result;        
}

File: solutions/dart/secret-handshake/1/lib/secret_handshake.dart
class SecretHandshake {
  List<String> commands(int code){
    List<String> actions = ["wink", "double blink", "close your eyes", "jump"];
    List<String> result = [];
    var test = 1;
    for(int i=0; i<4; i++){
      if(code & test != 0)
        result.add(actions[i]);
      test<<=1;
    }
    if(code & test != 0)
      result = result.reversed.toList();
    return result;
  }
}
